structured_data/2021_06_07_timeseries_labeling_tools/README.md

```diff
@@ -0,0 +1,282 @@
+- labels can be tagged with sub tags, and with a quality rating.
+- seems to have a nice community (albeit unsure how big the time series use case is)
+
+#### Setup
+There are different ways to install Label Studio.
+It can be easily installed using pip, using docker,... (https://labelstud.io/guide/install.html)
+
+When installing it in the cloud on Google Compute Engine, be aware of scopes to get it working:
+
+> Do take care of scopes! Those are a legacy layer on top of IAM. You need to give your CE instance the following settings:
+> - Cloud API access scopes
+> - Allow full access to all Cloud APIs
+
+By default an SQLite is used in the backend. 
+If this would be too limiting, a PostgreSQL database can be used as well (https://labelstud.io/guide/storedata.html). 
+
+In Label Studio users and projects can be created. 
+Within these projects different *label tasks* can be created. 
+Label tasks are time series segments that need to be labeled.
+Once the operator is happy with the labelling, it can be submitted.
+
+#### Import
+To import data, a data template template needs to be made. 
+This template is basically the configuration of how the data looks like (which columns are important), but also how the data can be viewed during labelling.
+This allows for a very flexible import system.
+More info on these data templates: [https://labelstud.io/templates/time_series.html](https://labelstud.io/templates/time_series.html)
+
+The data needs to be formatted in wide format, which for certain cases might lead to sparse datasets.
+On top of that, Label Studio seems to set missing values (~ sparse data) to 0, complicating importing data from a single source, for data that was not taken at the same time.
+To avoid this, multiple different files can be used.
+
+The tool can use different data sources. For an overview check: [https://labelstud.io/guide/storedata.html](https://labelstud.io/guide/storedata.html).
+As an example, Label Studio can directly connect to a GCP (storage bucket): [https://labelstud.io/guide/storage.html#Google-Cloud-Storage](https://labelstud.io/guide/storage.html#Google-Cloud-Storage).
+Unfortunately this did not seem to work outside of the VPC. This is probably due to a rights issue.
+
+Data can be imported using the GUI or using an API.
+Files up to 200Mb are supposed to work, although it started failing above 50Mb using the GUI.
+Files that are bigger suffer from a *validation error*.
+
+Label Studio has an API that can be used to pump data in.  ([https://labelstud.io/guide/tasks.html#Import-data-using-the-API](https://labelstud.io/guide/tasks.html#Import-data-using-the-API))
+
+Some other tips:
+- Data should be sorted on time
+- Missing data: data will be set to 0 (flat line).
+- Corrupt data (e.g. strings): time series just stop at the first time point it encountered a string. So even if the data afterwards is okay. There is simply no data to been seen (empty time series plot).
```

Comment:
```suggestion
- Corrupt data (e.g. strings): the data is imported correctly **until the first occurrence of a corrupt value**. Even with a single corrupt datapoint, this could mess up your whole dataset.
```